04week/algorithms.js

```diff
@@ -16,14 +16,56 @@ for (let i = 0; i < 1000; i++) {
 
 function bubbleSort(arr) {
   // Your code here
+  const ret = Array.from(arr)
+  let swapped
+  do {
+    swapped = false
+    for (let i = 1; i < ret.length; i++) {
+      if (ret[i - 1] > ret[i]) {
+        [ret[i], ret[i - 1]] = [ret[i] - 1, ret[i]]
+        swapped = true
+      }
+    }
+  } while (swapped);
+  return ret;
```

Comment:
I get where you're coming from, however I would like you to use a recursive function here, it will make this code a LOT cleaner, more efficient and less prone to errors.